pennylane/gradients/parameter_shift_hessian.py

```diff
@@ -181,6 +181,34 @@ def _generate_diag_tapes(tape, idx, diag_recipes, add_unshifted, tapes, coeffs):
     return add_unshifted, unshifted_coeff
 
 
+def _no_trainable_grad_new(tape):
+    if len(tape.measurements) == 1:
+        return [], lambda _: qml.math.zeros((0,))
+
+    return [], lambda _: tuple(qml.math.zeros((0,)) for _ in tape.measurements)
```

Comment:
The warning message is different (it explicitly mentions the hessian transform instead of a regular gradient transform). If you think this isn't an issue then I can use the same method as finite diff, but personally I'd like to keep them separate.

Reply:
I think if the logic at some point is duplicated, then it's best to be kept in a single place. The function can also be split into 1. warning 2. logic.

Not a major thing though.